python/TestHarness/TestHarness.py

```diff
@@ -370,6 +372,7 @@ def testOutputAndFinish(self, tester, retcode, output, start=0, end=0):
       (reason, output) = self.buildPBSBatch(output, tester)
     elif self.options.dry_run:
       reason = 'DRY_RUN'
+      test['status_message'] = reason
```

Comment:
I should be in the office tomorrow. I'll take a closer look with you and we can get this merged.